### PR TITLE
Build/update docker

### PR DESCRIPTION
**Pull request by vincepick**
Updated docker file for usage within AutoIG when running inside a container. 

Because changes were moved over to a fork, this will remain a draft until I have gone through and moved all my changes from all branches in the original repo into this PR, will be done soon.

### CODE REVIEW

**Comment by ndangtt** [comment]
@vincepick: thank you for the PR! I think it'd be best to fix those redundant installation first. It'd also be useful to test this docker and the updated installation scripts via running minizinc/essence experiments for all installed solvers.

**Comment by vincepick** [comment]
It is also worth noting that the **install-ortools** script can be deleted entirely, as the MZN bundle comes with ortools already installed, and the **update-or-path.sh** corrects its reference to still be used by AutoIG in the same way.

A call to **update-or-path script** was also added to the existing **set-path.sh** script, so even when not built in a container the program can still be run with the same sequence of commands as specified in the **read the docs**.  An alternative is to have the **update-or-path-script** called as a separate command call rather than in set path, to keep the **set-path** script more modular and only update the path variable. 

In the future, AutoIG can also be refactored to use the name for OR tools set in the MZN installation, which would prevent the need for this **update-or-path.sh** script to be used at all. But for now, this is a possible workaround to this problem and prevents the need for another instance of or-tools to be installed.

**Review by ndangtt — COMMENTED** [review]
@vincepick: thank you very much for the PR! I'd like to have one question: where are the two files `m.dzn` and `m.mzn` used in the `data/models/testModel/` folder used? Are there any tests that make use of them?

**Comment by vincepick** [comment]
I believe that those two models were just used during development and can be deleted.

File: DEV_README.md [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
-## To Run The Generated Bash Script
+`docker run --platform linux/amd64 -it -v <volume_name>:/PathToAutoIG <container-name> /bin/bash`
+
+# Commands To Run Inside Docker Container For Quick Start Example
+
+## Getting the environment ready: needs to be run every time the container is open
+
+`. bin/container-set-path.sh`
+
+`AUTOIG=$(pwd)`
```
Comment (ndangtt) [inline DEV_README.md:30]: we could change it to `export AUTOIG=$(pwd)`

File: DEV_README.md [hunk 2]
```diff
@@ -1,23 +1,49 @@
...
+
+`python $AUTOIG/scripts/setup.py --generatorModel $AUTOIG/data/models/macc/generator-small.essence --problemModel $AUTOIG/data/models/macc/problem.mzn --instanceSetting graded --minSolverTime 0 --maxSolverTime 5 --solver chuffed --solverFlags="-f" --maxEvaluations 180 --genSolverTimeLimit 5`
+
+### To Run The Generated Bash Script
 
 bash run.sh
+
+# Considerations for Use of Dockerfile
+
+The build Docker image allows for the program to be run in a container. It is worth noting though that the container could take up more storage than running AutoIG through Linux directly, as it will download dependencies within the container such as Python and R. If a users system already has these, it could be more efficient to run it directly on the system without a VM. In addition, data does not persist within the container, so it is important to save the results of AutoIG runs, perhaps with a Docker Volume. Instructions for setting up the Docker Volume are below.
```
Comment (ndangtt) [inline DEV_README.md:48]: instructions for setting the Docker Volume are listed **above**

**Comment by ndangtt** [comment]
@vincepick: the PR looks great to me. Many thanks again for your work on this, I've now merged the PR into dev :)

**Comment by vincepick** [comment]
Awesome, thank you for all your help in completing it!

**Pull request merged**